opaque_keys/edx/locator.py

```diff
@@ -151,6 +153,10 @@ class CourseLocator(BlockLocatorBase, CourseKey):   # pylint: disable=abstract-m
     __slots__ = KEY_FIELDS
     CHECKED_INIT = False
 
+    # A mapping of parsed key fields to CourseLocator objects, for caching.
+    # Every subclass has its own.
+    __parsed_fields_to_keys = WeakValueDictionary()
```

Comment:
How does every subclass get its own?  I only see the one definition, and thought the double underscore only does name mangling (all subclasses get the same mangled name and object reference, based on the class it was defined in).